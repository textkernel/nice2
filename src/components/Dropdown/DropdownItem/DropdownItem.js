import React from 'react';
import PropTypes from 'prop-types';
import bem from 'bem';
import Checkbox from '../../Checkbox';
import Text from '../../Text';
import { DropdownConsumer } from '../DropdownContext';
import styles from './DropdownItem.scss';

const { block, elem } = bem({
    name: 'DropdownItem',
    classnames: styles,
    propsToMods: []
});

const DropdownItem = props => {
    const { children } = props;

    return (
        <DropdownConsumer>
            {({ multiselect }) => (
                <div {...block(props)}>
                    {multiselect ? (
                        <div {...elem('multi', props)}>
                            <Checkbox>{children}</Checkbox>
                        </div>
                    ) : (
                        <div {...elem('single', props)}>
                            <Text inline>{children}</Text>
                        </div>
                    )}
                </div>
            )}
        </DropdownConsumer>
    );
};

DropdownItem.displayName = 'DropdownItem';

DropdownItem.propTypes = {
    disabled: PropTypes.bool
};

DropdownItem.defaultProps = {
    disabled: false
};

export default DropdownItem;
