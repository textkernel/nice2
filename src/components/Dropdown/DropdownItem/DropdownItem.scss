.DropdownItem {
    color: var(--color-foreground);
    cursor: pointer;
    display: block;
    line-height: var(--line-height-normal);
    padding: var(--spacing-2x) var(--spacing-3x);
    user-select: none;
    white-space: nowrap;

    &:hover {
        background-color: var(--color-neutral-25);
    }

    &--checkbox {
        padding: 0;
    }

    &--disabled {
        cursor: not-allowed;
        color: var(--color-neutral-75);

        &:active {
            background-color: var(--color-neutral-25);
        }
    }
    &--isStatic {
        background-color: var(--color-background);
        color: var(--color-neutral-75);
        cursor: default;

        &:hover,
        &:active {
            background-color: var(--color-background);
        }
    }

    &__checkbox {
        .Checkbox__label {
            padding: var(--spacing-2x) var(--spacing-3x);
            line-height: var(--line-height-normal);
        }
    }
}
