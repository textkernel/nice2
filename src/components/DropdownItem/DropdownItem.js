import React from 'react';
import PropTypes from 'prop-types';
import bem from '../../packages/bem';
import styles from './DropdownItem.scss';

const { block } = bem({
    name: 'DropdownItem',
    classnames: styles,
    propsToMods: ['disabled', 'isStatic']
});

const DropdownItem = props => {
    const { checkbox, children, disabled, isStatic, ...rest } = props;

    if (checkbox) {
        return (
            <div {...rest} {...block(props)}>
                <input type="checkbox" />
                <label>
                    {children}
                </label>
            </div>
        );
    }

    return (
        <div {...rest} {...block(props)}>
            {children}
        </div>
    );
};

DropdownItem.propTypes = {
    checkbox: PropTypes.bool,
    /** Item label */
    children: PropTypes.node.isRequired,
    /** Whether the item is disabled */
    disabled: PropTypes.bool,
    /** Callback function, fired when item is clicked */
    onClick: PropTypes.func,
    /** If true, this item does not allow user interaction */
    isStatic: PropTypes.bool,
    /** Selected value of this item */
    value: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
};

DropdownItem.defaultProps = {
    checkbox: false,
    disabled: false,
    onClick: null,
    isStatic: false,
    value: null
};

export default DropdownItem;
