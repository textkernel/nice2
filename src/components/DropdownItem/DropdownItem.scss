.DropdownItem {
    color: var(--color-foreground);
    display: block;
    padding: var(--spacing-2x) var(--spacing-3x);
    white-space: nowrap;

    &:hover {
        background-color: var(--color-neutral-25);
    }
}
