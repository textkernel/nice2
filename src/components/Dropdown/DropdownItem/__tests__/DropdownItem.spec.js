import React from 'react';
import toJson from 'enzyme-to-json';
import DropdownItem from '../DropdownItem';

describe('<DropdownItem> that renders a dropdown item', () => {
    it('should render a default item correctly', () => {
        const wrapper = shallow(<DropdownItem>A dropdown item</DropdownItem>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render a item with checkbox correctly', () => {
        const wrapper = shallow(<DropdownItem checkbox>A multiselect dropdown item</DropdownItem>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render a item with checked checkbox correctly', () => {
        const wrapper = shallow(
            <DropdownItem checkbox checked>
                A multiselect dropdown item
            </DropdownItem>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should add classes when props are changed', () => {
        const wrapper = shallow(<DropdownItem disabled>Another dropdown item</DropdownItem>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
