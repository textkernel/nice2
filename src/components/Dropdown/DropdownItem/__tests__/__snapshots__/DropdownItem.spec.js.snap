// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownItem> that renders a dropdown item should add classes when props are changed 1`] = `
<div
  className="DropdownItem DropdownItem--disabled"
  onClick={[Function]}
  role="menuitem"
>
  Another dropdown item
</div>
`;

exports[`<DropdownItem> that renders a dropdown item should render a default item correctly 1`] = `
<div
  className="DropdownItem"
  onClick={[Function]}
  role="menuitem"
>
  A dropdown item
</div>
`;

exports[`<DropdownItem> that renders a dropdown item should render a item with checkbox correctly 1`] = `
<div
  className="DropdownItem DropdownItem--checkbox"
>
  <Checkbox
    checked={false}
    className="DropdownItem__checkbox DropdownItem__checkbox--checkbox"
    disabled={false}
    id="item_undefined"
    onChange={[Function]}
    value={null}
  >
    A multiselect dropdown item
  </Checkbox>
</div>
`;

exports[`<DropdownItem> that renders a dropdown item should render a item with checked checkbox correctly 1`] = `
<div
  className="DropdownItem DropdownItem--checkbox"
>
  <Checkbox
    checked={true}
    className="DropdownItem__checkbox DropdownItem__checkbox--checkbox"
    disabled={false}
    id="item_undefined"
    onChange={[Function]}
    value={null}
  >
    A multiselect dropdown item
  </Checkbox>
</div>
`;
